### PR TITLE
jsx-no-target-blank to check for forms with target="_blank"

### LINKED ISSUE #1143: Add or modify rule for no target="_blank" to include form elements
```
      <form
        method="POST"
        action={url}
        target="_blank"
      >
          <formStuff />
      </form>
```

I could be wrong but isn't a form post open to the same vulnerabilities as a <a> with target="_blank".  The current implementation doesn't flag this.

### PR DESCRIPTION
**Pull request by jaaberg**
This should fix #1143.

I am not sure if this should be an option or not (as suggested in the issue)(?).

### CODE REVIEW

**Comment by jaaberg** [comment]
Added option `preventInForms` to opt-in to validating target-blank in forms with tests and docs.

**Comment by ljharb** [comment]
@jaaberg what about a schema like this:
```json5
{
  "forms": <boolean>, // default false
  "links": <boolean>, // default true
}
```

That way it's an easy change to flip the default from false to true in the next semver-major, and the schema is also extensible if we want to add other categories in the future?

**Comment by jaaberg** [comment]
@ljharb Thanks for feedback :) Updated the PR.

File: lib/rules/jsx-no-target-blank.js [hunk 1]
```diff
@@ -38,19 +41,40 @@ module.exports = {
...
     return {
       JSXAttribute: function(node) {
-        if (node.parent.name.name !== 'a') {
+        const config = Object.assign({
+          links: true,
+          forms: false
+        }, context.options[0]);
+
+        if (
+          (config.links ? node.parent.name.name !== 'a' : true) &&
```
Comment (ljharb) [inline lib/rules/jsx-no-target-blank.js]: `!config.links || node.parent.name.name !== 'a'`?

Comment (jaaberg) [reply lib/rules/jsx-no-target-blank.js]: Yeah I was thinking about it, do you think it is more readable?

Comment (ljharb) [reply lib/rules/jsx-no-target-blank.js]: hm, i'm not really sure. i'm not a fan of ternaries that select booleans, but as-is might be more readable in that case.

**Comment by ljharb** [comment]
Rebased, and added a global `formComponents` setting to go with the `linkComponents` setting that was added after this PR was created.

**Comment by ljharb** [comment]
@jaaberg i think this can land if it's rebased on top of the newly added `allowReferrer` and `warnOnSpreadAttributes` options.

**Review by ljharb — APPROVED** [review]

**Pull request merged**